src/components/elements/tableFilters/filters/ColumnsMenu.tsx

```diff
@@ -0,0 +1,132 @@
+import ViewColumnIcon from '@mui/icons-material/ViewColumn';
+import {
+  Box,
+  Button,
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  FormGroup,
+  Popover,
+  Stack,
+  Typography,
+} from '@mui/material';
+import { isEmpty } from 'lodash-es';
+import {
+  usePopupState,
+  bindTrigger,
+  bindMenu,
+} from 'material-ui-popup-state/hooks';
+import { ReactNode, useState } from 'react';
+
+import TableFilterButton from './FilterButton';
+
+export interface TableColumnsMenuProps {
+  columns: { value: string; header: ReactNode; defaultHidden: boolean }[];
+  columnsValue: string[];
+  setColumnsValue: (value: string[]) => any;
+}
+
+const TableColumnsMenu = ({
```

Comment:
The behavior doesn't quite match the "Filters" menu behavior. When you hit "Apply", the menu should close.